torch/csrc/stable/library.h

```diff
@@ -76,7 +85,21 @@ template <
     typename T,
     std::enable_if_t<!detail::is_optional<T>::value, bool> = true>
 T to(StableIValue val) {
-  return *reinterpret_cast<T*>(&val);
+  static_assert(std::is_trivially_copyable_v<T>);
+  // T may not have a default constructor. (For example, it might be
+  // c10::Device.) However, std::memcpy implicitly creates a T at the
+  // destination. So, we can use a union to work around this lack of
+  // default constructor.
+  union Result {
+    Result() {}
+    T t;
+  };
+  Result result;
+  // reinterpret_cast to suppress overzealous -Wclass-memaccess. We
+  // have a static_assert above that T is trivially copyable, which
+  // should be enough.
```

Comment:
nit: should we link https://gcc.gnu.org/bugzilla/show_bug.cgi?id=107361 in the comment here?